consoleApp/src/main/java/com/infoshare/database/FileDb.java

```diff
@@ -0,0 +1,112 @@
+    public void saveVolunteer(Volunteer volunteer) throws IOException {
+        // Sprawdzamy czy wolontariusz istnieje
+
+        if (getVolunteer(volunteer.getEmail()) == null) {
+            FileWriter fileWriter = new FileWriter(VOLUNTEER_DB_FILE_NAME, true);
+            fileWriter.write(volunteer.getName() + "," + volunteer.getLocation() + "," + volunteer.getEmail() + ","
+                    + volunteer.getPhone() + "," + volunteer.getTypeOfHelp() + "," + volunteer.isAvailable() + "\n");
+            fileWriter.close();
+        } else {
+            // wolontariusz istnieje wiec aktualizujemy jego dane
+
+            List<Volunteer> allVounteers = getVolunteers();
+            int index = -1;
+            for (int i = 0; i < allVounteers.size(); i++) {
+                Volunteer v = allVounteers.get(i);
+                if (v.getEmail().equals(volunteer.getEmail())) {
+                    index = i;
+                }
+            }
+            allVounteers.set(index, volunteer);  // gdy chce usunac .remove
+            FileWriter writer = new FileWriter(VOLUNTEER_DB_FILE_NAME, false);
+            for (Volunteer v : allVounteers) {
+                writer.write(v.getName() + "," + v.getLocation() + "," + volunteer.getEmail() + ","
+                        + v.getPhone() + "," + v.getTypeOfHelp() + "," + v.isAvailable() + "\n");
+            }
+            writer.close();
+        }
+    }
+
+    @Override// zapis zgłoszenia osob potrzebujacej pomocy wraz z rodzajem pomocy
+    public void saveNeedRequest(NeedRequest needRequest) throws IOException {
+        FileWriter fileWriter = new FileWriter(REQUEST_DB_FILE, true);
+        PersonInNeed person = needRequest.getPersonInNeed();
+        fileWriter.write(needRequest.getTypeOfHelp() + "," + needRequest.getHelpStatus() + ","
+                + df.format(needRequest.getStatusChange()) + ",");
+
+        fileWriter.write(person.getName() + "," + person.getLocation() + "," + person.getEmail()
+                + "," + person.getPhone() + "\n");
+        fileWriter.close();
+    }
+
+    //odczyt danych osoby potrzebujacej pomocy wraz z rodzajem pomocy
+    @Override
+    public List<NeedRequest> getAllNeedRequests() throws FileNotFoundException, ParseException {
+        List<NeedRequest> result = new ArrayList<>();
+        Scanner scanner = new Scanner(new File(REQUEST_DB_FILE));
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] splited = line.split(",");
+            PersonInNeed person = new PersonInNeed(splited[3], splited[4], splited[5], splited[6]);
+            NeedRequest needRequest = new NeedRequest(TypeOfHelp.valueOf(splited[0]), HelpStatuses.valueOf(splited[1])
+                    , df.parse(splited[2]), person);
+            result.add(needRequest);
+        }
+        return result;
+    }
+
+    @Override // odczyt dost. wolontariuszy , uznalem ze email jest unikatowy dla wolontariusza
```

Comment:
To wygląda bardziej na wyszukanie wolontariusza po mailu. Jeżeli miałby zwracać dostępnych wolontariuszy powinien zwracać listę i wybierać po isAvailable